### ISSUE TITLE
After packaging Electron app with Realm we get error from Realm library

### ISSUE THREAD

**Issue by bimusiek** [realm/realm-js#6303]
### How frequently does the bug occur?

Always

### Description

While in development mode (Electron with devserver), everything works as expected.
However, after the app is packed with electron-builder though, we get an error:
```
Uncaught TypeError: k.injectInjectables is not a function
    at 4875 (main.c4a42f82.js:formatted:2)
    at n (main.c4a42f82.js:formatted:2)
    at 713 (main.c4a42f82.js:formatted:2)
    at n (main.c4a42f82.js:formatted:2)
    at main.c4a42f82.js:formatted:2
    at main.c4a42f82.js:formatted:2
    at main.c4a42f82.js:formatted:2
```
![CleanShot%202023-12-05%20at%2016 10 35@2x](https://github.com/realm/realm-js/assets/1025588/ba7c9cf3-806e-4ca8-8700-3b0c6fae3d0a)

### Can you reproduce the bug?

Always

### Version

12.3.1

### What services are you using?

Atlas Device Sync

### Are you using encryption?

No

### Platform OS and version(s)

Electron 19.0.8

### Build environment

Which debugger for React Native: ..

**Comment by bimusiek** [comment]
Debuggin more, I have tried to add `asarUnpack` as in https://github.com/realm/realm-studio/blob/channel/major-14/package.json but it did not help.

I located `k` value:
![CleanShot 2023-12-05 at 19 47 42@2x](https://github.com/realm/realm-js/assets/1025588/a0c90633-45b8-44f2-9297-0bb985b66c15)

but not sure why is it a string.

**Comment by kraenhansen** [comment]
@bimusiek Thanks for reporting back on your progress here.

The main reason we use `asarUnpack` of the `realm.node` binary is for perform code-signing (via the `mac.binaries` property). I wonder how you're bundling you app's JS code? Perhaps you're using a WebPack configuration that resolves the `realm.node` module differently in dev vs prod mode?

**Comment by bimusiek** [comment]
With v11, the error is:
```
Uncaught TypeError: Cannot read properties of undefined (reading 'indexOf')
```
when it tries to import `realm.node`. Not sure why... but at least I found a duplicate:
https://github.com/realm/realm-js/issues/4741

However, there was no soluton for this issue. I see that Realm Studio uses `require` as a workaround, instead of `import Realm from 'realm'`.

**Comment by bimusiek** [comment]
@kraenhansen We use `craco` with `react-scripts`. 

For dev mode, we do: 'craco start' & `electron build/electron.js`.
For building we first package the app with `craco build` then use `electron-builder` to package everything together. 

And for both we use `webpack-node-externals`:
```js
webpackConfig.externals = [
    nodeExternals({
        allowlist: [/^(?!.*(^realm)$)/i],
        additionalModuleDirs: [path.resolve(__dirname, '../../../node_modules')],
    }),
]
```

Could it be too old version of Electron though? I am working on it for past few hours trying different things and different combinations of versions and solutions.

**Comment by bimusiek** [comment]
Ok, just checked without node-linker=hoisted, then dev server works correctly. But then electron-builder packages app incorrectly :D 

Basically, Realm + PNPM + Electron does not cooperate. 👎 
Switching build process to `yarn` 😢

**Comment by kraenhansen** [comment]
@bimusiek I agree - this is super unfortunate; there's a lot of libraries and tools which make assumptions on the relative paths of files and instead of relying on the node module resolution algorithm.

Please let us know if there's anything we can do or document to make Realm + Electron a smoother experience.

**Comment by kraenhansen** [comment]
Are you still applying this workaround? https://github.com/realm/realm-js/issues/3828#issuecomment-874655440

**Comment by bimusiek** [comment]
@kraenhansen If we will ever have a chance of meeting in real life, I owe you a beer! 👍 I have removed this workaround while debugging previous issue and it got lost in the set of changes. 

Thank you, not sure why I did not find it...

**Comment by kraenhansen** [comment]
Great to hear! We should probably just get that file included in the package ourselves. It's just very strange that this is the way Electron builder detects prebuilds and I'd love to find a less arbitrary solution 🤔

**Comment by bimusiek** [comment]
Just a polish idiom for you, which translate roughly to: "Better is an enemy of good" 😄 

Having just a workaround for anyone that tries Realm with Electron for the first time can be beneficial. Even if it is just a workaround. And then after, there is time to ask, how are we gonna do it right 🔎

**Comment by kraenhansen** [comment]
You're right .. I merged the workaround 👍
